### PR TITLE
[paused] feat: snos job

### PR DESCRIPTION
**Pull request by akhercha**
## ⏸ On hold

This PR is currently paused until the [snos](https://github.com/keep-starknet-strange/snos) repository has been updated (since the way to run Snos in the context of Madara will change a lot and we don't have any spec/design yet).

________

Resolves: #11 

## Added

- Implementation of the `SnosJob`
- `StateJob::DummyState` - used as a replacement until we can import the structure `BlockifierStateAdapter` from Madara. Currently not possible dues to conflicting dependencies versions between Snos <=> Madara <=> Cairo-vm. (It may not be needed at all to import that depending on the next updates from snos, their goal is to make it easier to run the Snos in the context of Madara)
- `HttpRpcClient` to our `Config` for RPC requests to L1/Madara (gas price & snos input at the moment)
- `conversions` & `time` utils functions

### CODE REVIEW

File: crates/orchestrator/src/jobs/snos_job/mod.rs [hunk 1]
```diff
@@ -29,28 +51,133 @@ impl Job for SnosJob {
...
 
     fn max_process_attempts(&self) -> u64 {
-        todo!()
+        1
     }
 
     fn max_verification_attempts(&self) -> u64 {
-        todo!()
+        // TODO: isn't 10 a lot?
+        10
```
Comment (apoorvsadana) [inline crates/orchestrator/src/jobs/snos_job/mod.rs:148]: ya if there's no verification then 10 would be a lot :)

File: crates/orchestrator/src/jobs/snos_job/mod.rs [hunk 2]
```diff
@@ -29,28 +51,133 @@ impl Job for SnosJob {
...
     }
 
     fn verification_polling_delay_seconds(&self) -> u64 {
-        todo!()
+        // TODO: what is an average run time for SNOS?
+        60
```
Comment (apoorvsadana) [inline crates/orchestrator/src/jobs/snos_job/mod.rs:153]: we don't know the runtime yet but in the current implementation looks like we run in synchronously so verification delay can be like 1s

Comment (akhercha) [reply crates/orchestrator/src/jobs/snos_job/mod.rs]: Ah yes true - updated

**Comment by Tranduy1dol** [comment]
Hi @apoorvsadana , I just want to know if there are any plans for SNOS. I've noticed that this PR has been on hold for a while.

**Comment by akhercha** [comment]
> Hi @apoorvsadana , I just want to know if there are any plans for SNOS. I've noticed that this PR has been on hold for a while.

Hey @Tranduy1dol - this PR is on hold until the `snos` repository is ready to be integrated with Madara. Until then, we can't really continue this work.

**Pull request closed without merging**